### ISSUE TITLE
Please add toprettyxml

### ISSUE THREAD

**Issue by mtthws** [pabigot/pyxb#25]
Hi,
I recently started using your library.  It is very nice, thanks a lot.

Would it be possible to add toprettyxml in addition to toxml?

Randomness, is there a reason in six.py line 482 the param passed is triple quoted instead of single quoated?

Also structures.py line 1753 has a messed up doc string.  It is "doc""" instead of """doc"""

Thanks for such a nice library
have fun

**Comment by zyv** [comment]
I second that `toprettyxml()` would be nice to have!

**Comment by pabigot** [comment]
After considering this carefully I'm going to decline the enhancement.  The problem is that the existing `toxml()` method combines arguments meant to control the conversion of the PyXB instance to a DOM structure, and to control the conversion of the DOM structure to XML text.  It has only one argument that affects the DOM-to-XML step, and is essentially mandatory, so fits naturally as the first argument with the instance-to-DOM arguments which are rarely used following it.

In contrast `toprettyxml()` introduces two additional options specific to the DOM-to-XML phase, and places them before the mandatory `encoding` argument, which would make correct usage difficult when instance-to-DOM arguments are needed.

In short, at this point I sorta regret adding `toxml()` as a short-hand in the first place, so am sticking to my original response that `x.toDOM().toprettyxml()` is the recommended path.

**Comment by zyv** [comment]
Maybe one could add an optional argument to `toxml()` as in `pretty=False` instead or that'd be problematic in some other way? Just wondering if you have also considered that...

**Comment by pabigot** [comment]
@zyv In fact, that's exactly how I started.  But then, for `pretty=True` to be fully useful, I'd have to also add the `indent` and `newl` arguments which as I noted are in an inconsistent order relative to the existing `encoding` argument so they'd have to be keyword arguments too as there are already four parameters to `toxml()`.  At that point I noticed that both functions are only available in [minidom](https://docs.python.org/2/library/xml.dom.minidom.html) and PyXB isn't supposed to be restricted to a particular DOM implementation.

So trying to graft it in just got too messy: the best solution IMO is to convert to DOM, then use the native functions provided by the DOM implementation.
